distributor/distributor.go

```diff
@@ -330,6 +331,11 @@ func (d *Distributor) Push(ctx context.Context, req *remote.WriteRequest) (*cort
 	}
 	for i := range sampleTrackers {
 		if sampleTrackers[i].succeeded < int32(sampleTrackers[i].minSuccess) {
+			if lastErr == util.ErrUserSeriesLimitExceeded {
+				// We can't wrap this error or it won't be communicated back to
+				// the user correctly.
+				return nil, lastErr
+			}
```

Comment:
In #272 I get rid of the custom error below and just propagate the error from the ingester (so compatible with this change).

I do need to give a bit of a think about how this throttling will interact with the quorum writes though...